client/v2/autocli/msg.go

```diff
@@ -217,11 +216,67 @@ func (b *Builder) handleGovProposal(
 	// Here we use dynamicpb, to create a proto v1 compatible message.
 	// The SDK codec will handle protov2 -> protov1 (marshal)
 	msg := dynamicpb.NewMessage(input.Descriptor())
-	proto.Merge(msg, input.Interface())
+	cloneMessage(msg, input)
 
 	if err := proposal.SetMsgs([]gogoproto.Message{msg}); err != nil {
 		return fmt.Errorf("failed to set msg in proposal %w", err)
 	}
 
 	return clienttx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), proposal)
 }
+
+// cloneMessage safely copies fields from src message to dst message.
+// this avoids the proto.Merge issue with field descriptors from different repositories.
+func cloneMessage(dst, src protoreflect.Message) {
```

Comment:
we need to test this method

Reply:
i'll add a test 👍🏾

Reply:
Added a test: https://github.com/cosmos/cosmos-sdk/pull/24722/commits/6bdfb6510e6c7b9e23c7a03f8f1c43eda025478c.
It fails before with that error, passes after the fix.

Reply:
I was thinking of the hacks needed to get the msg url. Let me try to directly use that dynamicpb message

Reply:
But you shouldn't be using gogo proto unmarshal here. It should just be proto unmarshal because that's what was used to dynamically construct msg

Reply:
yes, gogoproto unmarshal fallbacks to that: https://github.com/cosmos/gogoproto/blob/main/proto/decode.go#L344-L346

Reply:
The failure comes from the `TestMsgGogo` and `TestMsg` if you apply the snippet from above.
It fails because it the result doesn't match the golden files (and the golden files pushed to this branch are correct)

Reply:
The code you're showing is incorrect. You're not copying input to msg at all and you're calling the wrong marshal methods. You don't have proto.Message to begin with. Look at the commit that I linked above

Reply:
🤦‍♂️ lol my bad. I'll try it later and update it if the tests are passing. Thanks